plutus-use-cases/src/Plutus/Contracts/Auction.hs

```diff
@@ -130,7 +132,9 @@ auctionTransition AuctionParams{apOwner, apAsset, apEndTime} State{stateData=old
                 newState =
                     State
                         { stateData = Ongoing HighestBid{highestBid = newBid, highestBidder = newBidder}
-                        , stateValue = apAsset <> Ada.toValue newBid -- and lock the new bid in the script output
+                        , stateValue = Value.noAdaValue oldStateValue
+                                    <> Ada.toValue (Ada.fromValue oldStateValue - highestBid)
+                                    <> Ada.toValue newBid -- and lock the new bid in the script output
```

Comment:
This new stateValue calculation is more precise, and works better with the StateMachine now that it adjustes the Ada per tx output.